examples/MomentoRedisLambdaExample/MomentoRedisExampleLambdaHandler/src/MomentoRedisExampleLambdaHandler/Function.cs

```diff
@@ -0,0 +1,153 @@
+        
+        return "Task completed";
+    }
+
+    private async Task<string> DoStuff(Input input) {
+
+        Console.WriteLine("Start executing DoStuff....");
+
+        // Create Momento client
+        ICredentialProvider authProvider = new EnvMomentoTokenProvider("MOMENTO_AUTH_TOKEN");
+        var momentoClient = new CacheClient(
+            Configurations.InRegion.Lambda.Latest().WithClientTimeout(TimeSpan.FromSeconds(3000)),
+            authProvider,
+            TimeSpan.FromSeconds(30)
+        );
+        
+        // Create a Redis client
+        Console.WriteLine("Creating redis client");
+        var configurationOptions = new ConfigurationOptions
+        {
+            EndPoints = { RedisClusterEndpoint }
+        };
+        ConnectionMultiplexer redis = ConnectionMultiplexer.Connect(configurationOptions);
+        
+        // Instantiate MomentoCache and RedisCache
+        MomentoCache momentoCache = new MomentoCache(momentoClient, CacheName);
+        RedisCache redisCache = new RedisCache(redis);
+
+        // Create momento cache if not already exists
+        await momentoCache.CreateMomentoCache();
+        
+        // Ping redis client
+        redisCache.PingRedis();
+        
+        await ExecuteLambda(input, momentoCache, redisCache);
+        
+        // Close the redis connection
+        await redis.CloseAsync();
+        
+        Console.WriteLine("End executing DoStuff");
+
+        return "Task completed!";
+    }
+
+    private async Task<string> ExecuteLambda(Input input, MomentoCache momentoCache, RedisCache redisCache)
+    {
+        Console.WriteLine("Inside Execute Lambda function");
+        Console.WriteLine($"Input {input}");
+
+        List<Task<SetResult>> momentoSetTasks;
+        List<Task<GetResult>> momentoGetTasks;
+        List<Task<RedisSetResult>> redisSetTasks;
+        List<Task<RedisGetResult>> redisGetTasks;
+        
+        momentoSetTasks = input.RunMomento ? await momentoCache.DoMomentoSets(_numMomentoSetRequests, withSleep: input.SleepMomento) : new List<Task<SetResult>>();
+        redisSetTasks = input.RunRedis ? await redisCache.DoRedisSets(_numRedisSetRequests) : new List<Task<RedisSetResult>>();
+        momentoGetTasks = input.RunMomento ? await momentoCache.DoMomentoGets(_numMomentoGetRequests, compressValue: input.CompressMomento, withSleep: input.SleepMomento) : new List<Task<GetResult>>();
+        redisGetTasks = input.RunRedis ? await redisCache.DoRedisGets(_numRedisGetRequests, compressValue: input.CompressRedis) : new List<Task<RedisGetResult>>();
+
+        Console.WriteLine("Awaiting pending tasks");
+        // empty lists should no-op
+        Task.WaitAll(momentoSetTasks.Select(task => (Task)task).ToArray());
+        Task.WaitAll(redisSetTasks.Select(task => (Task)task).ToArray());
+        Task.WaitAll(momentoGetTasks.Select(task => (Task)task).ToArray());
+        Task.WaitAll(redisGetTasks.Select(task => (Task)task).ToArray());
```

Comment:
what happens if you don't have the `Select`s here? I wouldn't have expected those to be necessary.